### PR TITLE
ENH: Support CustomBusinessHour

### LINKED ISSUE #11514: Mixing CustomBusinessDay and BusinessHours Classes
Is it currently possible to combine the CustomBusinessDay and BusinessHour classes in any way?

My use-case is that I want to be able to create a SubClass of BusinessHour that would skip over all time occurring in the 'holidays' or outside the 'weekmask' defined within a CustomBusinessDay instance. This would be incredibly useful for me, and, I suspect, many others.

I have currently tried subclassing BusinessHour and replacing the next_bday attribute with in an instance of a CustomBusinessDate. Unfortunately, it does not rollforward past days defined as holidays, and seems to function the same as a typical BusinessDay object.

If such functionality is not currently available, please consider this a feature request.

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] closes #11514
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. pls rebase

**Comment by sinhrks** [comment]
Rebased, and now green.

File: doc/source/timeseries.rst [hunk 1]
```diff
@@ -883,6 +884,40 @@ under the default business hours (9:00 - 17:00), there is no gap (0 minutes) bet
...
+
+.. versionadded:: 0.18.1
+
+The ``CustomBusinessHour`` is a mixture of ``BusinessHour`` and ``CustomBusinessDay`` which
+allows you to specify arbitrary holidays. ``CustomBusinessHour`` works as the same
+as ``BusinessHour`` except that it skips specified custom holidays.
+
+.. ipython:: python
+
+    from pandas.tseries.holiday import USFederalHolidayCalendar
```
Comment (jreback) [inline doc/source/timeseries.rst]: can do this on another PR, but I think we need to be explict about these mimports (e.g. ``pd.offsets.CustomeBusinessHour``)
